### PR TITLE
Gpu access

### PR DESCRIPTION
**Pull request by dtmoodie**

### CODE REVIEW

File: src/operator/contrib/multibox_detection.cu [hunk 1]
```diff
@@ -200,7 +200,7 @@ inline void MultiBoxDetectionForward(const Tensor<gpu, 3, DType> &out,
   const int num_classes = cls_prob.size(1);
   const int num_anchors = cls_prob.size(2);
   const int num_batches = cls_prob.size(0);
-  const int num_threads = cuda::kMaxThreadsPerBlock;
+  const int num_threads = 256;
```
Comment (piiswrong) [inline src/operator/contrib/multibox_detection.cu]: ?

File: cpp-package/include/mxnet-cpp/MxNetCpp.h [hunk 2]
```diff
@@ -15,7 +15,7 @@
 #include "mxnet-cpp/operator.hpp"
 #include "mxnet-cpp/optimizer.hpp"
 #include "mxnet-cpp/kvstore.hpp"
-#include "mxnet-cpp/op.h"
+#include "mxnet-cpp/operator.h"
```
Comment (piiswrong) [inline cpp-package/include/mxnet-cpp/MxNetCpp.h]: ?

File: cpp-package/include/mxnet-cpp/ndarray.hpp [hunk 3]
```diff
@@ -331,6 +332,16 @@ inline const mx_float *NDArray::GetData() const {
   return static_cast<mx_float*>(ret);
 }
 
+inline const mx_float *NDArray::GetDataGpu() const {
+  void *ret;
+  CHECK_EQ(GetContext().GetDeviceType(), DeviceType::kGPU);
```
Comment (piiswrong) [inline cpp-package/include/mxnet-cpp/ndarray.hpp]: I don't think you need a new interface. You can simply remove the check above.

@lx75249

Comment (dtmoodie) [inline cpp-package/include/mxnet-cpp/ndarray.hpp]: I believe I would have to change MXNDArrayGetData to remove the check from the C api, which may cause segfaults if not properly checked in the other interfaces.

**Comment by piiswrong** [comment]
@lx

**Comment by dtmoodie** [comment]
Updated.

**Pull request closed without merging**